agents/craftassist/craftassist_swarm_worker.py

```diff
@@ -0,0 +1,345 @@
+        self.set_held_item = self.cagent.set_held_item
+        self.step_pos_x = self.cagent.step_pos_x
+        self.step_neg_x = self.cagent.step_neg_x
+        self.step_pos_z = self.cagent.step_pos_z
+        self.step_neg_z = self.cagent.step_neg_z
+        self.step_pos_y = self.cagent.step_pos_y
+        self.step_neg_y = self.cagent.step_neg_y
+        self.step_forward = self.cagent.step_forward
+        self.look_at = self.cagent.look_at
+        self.set_look = self.cagent.set_look
+        self.turn_angle = self.cagent.turn_angle
+        self.turn_left = self.cagent.turn_left
+        self.turn_right = self.cagent.turn_right
+        self.place_block = self.cagent.place_block
+        self.use_entity = self.cagent.use_entity
+        self.use_item = self.cagent.use_item
+        self.use_item_on_block = self.cagent.use_item_on_block
+        self.is_item_stack_on_ground = self.cagent.is_item_stack_on_ground
+        self.craft = self.cagent.craft
+        self.get_blocks = self.cagent.get_blocks
+        self.get_local_blocks = self.cagent.get_local_blocks
+        # self.get_incoming_chats = self.get_chats
+        self.get_player = self.cagent.get_player
+        self.get_mobs = self.cagent.get_mobs
+        self.get_other_players = self.get_all_players
+        self.get_other_player_by_name = self.cagent.get_other_player_by_name
+        self.get_vision = self.cagent.get_vision
+        self.get_line_of_sight = self.cagent.get_line_of_sight
+        self.get_player_line_of_sight = self.get_all_player_line_of_sight
+        self.get_changed_blocks = self.cagent.get_changed_blocks
+        self.get_item_stacks = self.cagent.get_item_stacks
+        self.get_world_age = self.cagent.get_world_age
+        self.get_time_of_day = self.cagent.get_time_of_day
+        self.get_item_stack = self.cagent.get_item_stack
+
+
+    def add_self_memory_node(self):
+        """Adds agent node into its own memory"""
+        # how/when to, memory is initialized before physical interfaces...
+        try:
+            p = self.get_player()
+        except:  # this is for test/test_agent
+            return
+        PlayerNode.create(self.memory, p, memid=self.memory.self_memid)
+
+
+if __name__ == "__main__":
+    base_path = os.path.dirname(__file__)
+    parser = ArgumentParser("Minecraft", base_path)
+    opts = parser.parse()
+
+    logging.basicConfig(level=opts.log_level.upper())
+
+    # set up stdout logging
+    sh = logging.StreamHandler()
+    sh.setFormatter(log_formatter)
+    logger = logging.getLogger()
+    logger.addHandler(sh)
+    logging.info("LOG LEVEL: {}".format(logger.level))
+
+    # Check that models and datasets are up to date and download latest resources.
+    # Also fetches additional resources for internal users.
+    if not opts.dev:
+        rc = subprocess.call([opts.verify_hash_script_path, "craftassist"])
+
+    set_start_method("spawn", force=True)
+
+    sa = CraftAssist_SwarmWorker(opts)
```

Comment:
Overall, instead of copying over the `craftassist_agent` files, wouldn't it be easier if we made:
`CraftAssistSwarmWorker` and `CraftAssistSwarmMaster` be classes that derive from `CraftAssistAgent` class ?
That way, we can just add / override the relevant methods.